samples/bluetooth/scanning_while_connecting/central/README.rst

```diff
@@ -0,0 +1,238 @@
+Once the connection establishment procedure is complete, it can immediately initiate a new connection establishment procedure to the cached device.
+When connecting to a cached device, the connection establishment procedure takes one less advertising interval compared to the sequential scanning and connection establishment mode.
+
+.. msc::
+    hscale = "1.3";
+    App, Stack, Peers;
+    App=>Stack [label="scan_start()"];
+    Peers=>Stack [label="ADV_IND(A)"];
+    Stack=>App [label="scan_recv(A)"];
+    App=>Stack [label="bt_conn_le_create(A)"];
+    Peers=>Stack [label="ADV_IND(B)"];
+    App rbox App [label="Cache address B"];
+    Peers=>Stack [label="ADV_IND(A)"];
+    Stack=>Peers [label="CONNECT_IND(A)"];
+    Stack=>App [label="connected_cb(A)"];
+    App=>Stack [label="bt_conn_le_create(B)"];
+    Peers=>Stack [label="ADV_IND(B)"];
+    Stack=>Peers [label="CONNECT_IND(B)"];
+    Stack=>App [label="connected_cb(B)"];
+
+Concurrent scanning while connecting with the filter accept list
+================================================================
+
+This mode requires the application to enable :kconfig:option:`CONFIG_BT_FILTER_ACCEPT_LIST` in addition to :kconfig:option:`CONFIG_BT_SCAN_AND_INITIATE_IN_PARALLEL`.
+When the application starts the connection establishment procedure with the filter accept list, it can connect to any of the previously cached devices.
+This reduces the total connection setup time even more, because connection establishment is not relying on the on-air presence of only one of the cached devices.
+
+.. msc::
+    hscale = "1.3";
+    App, Stack, Peers;
+    App=>Stack [label="scan_start()"];
+    Peers=>Stack [label="ADV_IND(A)"];
+    Stack=>App [label="scan_recv(A)"];
+    App=>Stack [label="bt_conn_le_create(A)"];
+    Peers=>Stack [label="ADV_IND(B)"];
+    Peers=>Stack [label="ADV_IND(C)"];
+    Peers=>Stack [label="ADV_IND(D)"];
+    App rbox App [label="Cache addresses B, C, D"];
+    Peers=>Stack [label="ADV_IND(A)"];
+    Stack=>Peers [label="CONNECT_IND(A)"];
+    Stack=>App [label="connected_cb(A)"];
+    App rbox App [label="Set filter accept list to\nB, C, D"];
+    App=>Stack [label="bt_conn_le_create_auto()"];
+    Stack rbox Stack [label="The stack will connect to the first present ADV_IND of\nthe peers B,C,D"];
+    Peers=>Stack [label="ADV_IND(C)"];
+    Stack=>Peers [label="CONNECT_IND(C)"];
+    Stack=>App [label="connected_cb(C)"];
+
+.. note::
+   This sample application assumes it will never have to cache more devices than the maximum number of addresses that can be stored in the filter accept list.
+   For applications that cannot adhere to this simplification, the function :cfunc:`cache_peer_address` can be changed to not store more than :kconfig:option:`CONFIG_BT_CTLR_FAL_SIZE`.
+   Another simplification done in the sample application is storing duplicate devices in the filter accept list.
+
+Configuration
+*************
+
+|config|
+
+Configuration options
```

Comment:
No need for this section unless you have some sample-specific new Kconfig options that are not available in the Kconfig reference. The one here is already available.